docs: add documentation of 5.4 features (#83)
neo4j/docs-spark: modules/ROOT/pages/types.adoc

|`Date`
|`date`
|
|Example: `2020-09-11`

|`Time`
|`struct { type: string, value: string }`
|
|Example: `[offset-time, 12:14:08.209Z]`

|`LocalTime`
|`struct { type: string, value: string }`
|
|Example: `[local-time, 12:18:11.628]`

|`ZonedDateTime`
|`timestamp`
|v5.4.0*
|Example: `2020-09-11 12:17:39.192+02:00`
Spark timestamps (`TIMESTAMP`) are written as `ZonedDateTime` in UTC. Prior to 5.4.0. these timestamps are written as `LocalDateTime`. This behavior can be restored after 5.4.0 with the setting `type.conversion` set to `legacy`.

|`LocalDateTime`
|`timestamp_ntz`
|v5.4.0*
|Example: `2020-09-11 12:14:49.081`
Spark local timestamps (`TIMESTAMP_NTZ`) are written as `LocalDateTime`. Prior to 5.4.0. these timestamps are written as a numerical value. This behavior can be restored after 5.4.0 with the setting `type.conversion` set to `legacy`.

|`Duration`
|`struct { type: string, months: long, days: long, seconds: long, nanonseconds: integer, value: string }`
|
|See link:https://neo4j.com/docs/cypher-manual/current/values-and-types/temporal/#cypher-temporal-durations[Temporal functions: duration].

|`Duration`
|`duration` or `period` objects
|v5.4.0*

> |Example: `java.time.Duration.ofDays(42)` or `java.time.Period.ofMonths(5)`. Prior to 5.4.0, Spark durations and intervals are stored as long. This behavior can be restored after 5.4.0 with the setting `type.conversion` set to `legacy`.

Reviewer: ```suggestion
|Example: `java.time.Duration.ofDays(42)` or `java.time.Period.ofMonths(5)`. 
Prior to 5.4.0, Spark durations and intervals are stored as `long`.
This behavior can be restored after 5.4.0 by setting `type.conversion` to `legacy`.
```
